libimage/pull.go

```diff
@@ -516,6 +592,9 @@ func (r *Runtime) copySingleImageFromRegistry(ctx context.Context, imageName str
 	}
 
 	sys := r.systemContextCopy()
+	if options.shortNameMode != nil {
+		sys.ShortNameMode = options.shortNameMode
+	}
 	resolved, err := shortnames.Resolve(sys, imageName)
```

Comment:
I’d expect the `PullCandidates` call to return exactly the values from this `Resolve`, i.e. the options that would be used for local prompting.

This PR instead filters them through `PullPolicyNewer`; is that necessary?

[Also, that’s a pretty deep call stack from `PullCandidates` all the way here. I guess it’s necessary, and it’s definitely better to share the call stack than to maintain a completely parallel `PullCandidates` code path — I’m just intuitively, with no knowledge of the code at all, tempted to go looking for ways to make that simpler.]

Reply:
Thanks for checking (and spotting!). The filtering of `PullPolicyNewer` was intended but is wrong :^)  As you pointed out, the local podman is prompting.

Regarding the call stack:  I was tempted to just call `shortnames.Resolve` but then started to add all the special cases where we are actually no prompting, most importantly, when resolving to a local image via ID, digest or with `PullPolicyNever`.  I found the code share to be easier to maintain in the future.